taipy/core/_orchestrator/_orchestrator.py

```diff
@@ -182,23 +170,22 @@ def _orchestrate_job_to_run_or_block(cls, jobs: List[Job]):
             cls.jobs_to_run.put(job)
 
     @classmethod
-    def __wait_until_job_finished(cls, jobs: Union[List[Job], Job], timeout: Optional[Union[float, int]] = None):
-        def __check_if_timeout(start, timeout):
-            if timeout:
-                return (datetime.now() - start).seconds < timeout
+    def _wait_until_job_finished(cls, jobs: Union[List[Job], Job], timeout: Optional[Union[float, int]] = None):
```

Comment:
🎨 Note related to the PR: `Union[float, int]` can be replaced by `float` because `int` is implied in the type hint `float`. See https://peps.python.org/pep-0484/#the-numeric-tower.